packages/dashboard/src/lib/components/shared/universal-input-components.tsx

```
@@ -0,0 +1,229 @@
+        const max = fieldDef.ui?.max;
+        const step = fieldDef.ui?.step || (isFloat ? 0.01 : 1);
+        const prefix = fieldDef.ui?.prefix;
+        const suffix = fieldDef.ui?.suffix;
+
+        // Use AffixedInput if we have prefix/suffix or are in config args mode
+        if (prefix || suffix || valueMode === 'json-string') {
+            const numericValue = transformedValue !== undefined && transformedValue !== '' 
+                ? (typeof transformedValue === 'number' ? transformedValue : parseFloat(transformedValue) || '') 
+                : '';
+
+            const handleNumericChange = React.useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)